lib/OpenQA/Schema/Result/JobModules.pm

```diff
@@ -124,22 +124,29 @@ sub results {
     # load detail file which restores all results provided by os-autoinst (with hash-root)
     # support also old format which only restores details information (with array-root)
     my $results = ref($ret) eq 'HASH' ? $ret : {details => $ret};
+    my $details = $results->{details};
 
     # when the job module is running, the content of the details file is {"result" => "running"}
     # so set details to []
-    if (!$results->{details} || ref($results->{details}) ne "ARRAY") {
+    if (ref $details ne 'ARRAY') {
         $results->{details} = [];
         return $results;
     }
 
-    for my $step (@{$results->{details}}) {
-        if ($collect_text && $step->{text} && !defined($step->{text_data})) {
-            my $file = path($dir, $step->{text});
-            $step->{text_data} = decode('UTF-8', $file->slurp) if -e $file;
+    for my $step (@$details) {
+        my $text_file_name = $step->{text};
+        if ($text_file_name && !defined $step->{text_data}) {
```

Comment:
If you load the uncollected text files unconditionally, you'll massively slow down job status updates sent by the worker.

Reply:
I'll check when this route/function is being called. If it is on every job status update then this would be bad, indeed. But then an explaining comment would be useful. When I've read your code I've assumed you just wanted to workaround the failing jobs API test.

Reply:
Final job result calculation is cycling through all `$module->result` data and doesn't need the text files. With unconditional loading, marking ltp_openposix jobs on OSD as finished could take up to a minute.

Reply:
I could not find any occurrence where `OpenQA::Schema::Result::JobModules::results` is used besides the two usages which you've amended to *pass* the parameter for collecting (`$module->results(1);` in `read_test_modules` and `$module->results(1)->{details};` in `check_tabmode`) and the usage in the tests. I've also been tracking the code paths executed on a worker status update and could not find any usage of the results function. Am I missing something here?

Reply:
Yes, you're missing `Schema::Result::Jobs::_failure_reason()` which is indirectly called from `Schema::Result::Jobs::done()`.
https://github.com/os-autoinst/openQA/blob/master/lib/OpenQA/Schema/Result/Jobs.pm#L1620

Reply:
Right. It is however not called on every job status update then. Nevertheless it makes likely sense to skip collecting the text files in this case.